File: ios/Classes/AppleLiquidSymbolWeight.swift
import SwiftUI
import UIKit

enum AppleLiquidSymbolWeight {
  static func imageWeight(_ value: String?) -> UIImage.SymbolWeight? {
    switch value {
    case "ultraLight":
      return .ultraLight
    case "thin":
      return .thin
    case "light":
      return .light
    case "regular":
      return .regular
    case "medium":
      return .medium
    case "semibold":
      return .semibold
    case "bold":
      return .bold
    case "heavy":
      return .heavy
    case "black":
      return .black
    default:
      return nil
    }
  }

  static func fontWeight(_ value: String?) -> Font.Weight? {
    switch value {
    case "ultraLight":
      return .ultraLight
    case "thin":
      return .thin
    case "light":
      return .light
    case "regular":
      return .regular
    case "medium":
      return .medium
    case "semibold":
      return .semibold
    case "bold":
      return .bold
    case "heavy":
      return .heavy
    case "black":
      return .black
    default:
      return nil
    }
  }
}

File: lib/src/apple_liquid_symbol_weight.dart
/// SF Symbol stroke weights supported by Apple's symbol configuration APIs.
enum AppleLiquidSymbolWeight {
  /// The thinnest available SF Symbol stroke.
  ultraLight('ultraLight', 100),

  /// A very thin SF Symbol stroke.
  thin('thin', 100),

  /// A light SF Symbol stroke.
  light('light', 300),

  /// The regular SF Symbol stroke.
  regular('regular', 400),

  /// A medium SF Symbol stroke.
  medium('medium', 500),

  /// A semibold SF Symbol stroke.
  semibold('semibold', 600),

  /// A bold SF Symbol stroke.
  bold('bold', 700),

  /// A heavy SF Symbol stroke.
  heavy('heavy', 800),

  /// The heaviest available SF Symbol stroke.
  black('black', 900);

  const AppleLiquidSymbolWeight(this.platformValue, this.fallbackIconWeight);

  /// String value understood by the native iOS implementation.
  final String platformValue;

  /// Approximate Flutter `Icon.weight` value for non-iOS fallbacks.
  final double fallbackIconWeight;
}
